$OpenBSD: patch-src_main_c,v 1.12 2012/09/18 12:35:03 dcoppa Exp $

Fix history memory leak
With 'history = 0' playlist items are discarded, but not freed
(upstream git commit e3d8f497baaf02daeddb060cd986681d0ee208de)

--- src/main.c.orig	Tue Sep 18 14:22:12 2012
+++ src/main.c	Tue Sep 18 14:22:53 2012
@@ -301,6 +301,7 @@ static void BarMainLoop (BarApp_t *app) {
 			if (app->playlist != NULL) {
 				PianoSong_t *histsong = app->playlist;
 				app->playlist = app->playlist->next;
+				histsong->next = NULL;
 				BarUiHistoryPrepend (app, histsong);
 			}
 			if (app->playlist == NULL) {
